runtime/src/accounts_db.rs

```diff
@@ -9384,6 +9391,13 @@ impl AccountsDb {
     }
 }
 
+/// Specify the source of the accounts data when calculating the accounts hash
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum CalcAccountsHashDataSource {
+    Index,
```

Comment:
`IndexForTests`

This option should not be used by the real validator. This code path could be deleted completely except it is useful as a check on the account index, clean, and the storage based calculation.

Reply:
it is confusing, but it would be nice if we're documenting it that somehow we indicate that it is for debugging or tests. A comment is better than nothing.